src/scripts/main.js

```diff
@@ -1,11 +1,30 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    const handler = () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+      element.removeEventListener(eventName, handler);
+    };
+
+    element.addEventListener(eventName, handler);
+  });
```

Comment:
The `waitFor` function correctly sets up a promise that resolves when the specified event occurs on the element. However, ensure that the elements (`loginField`, `passwordField`, `button`) exist in the DOM before calling `waitFor`, otherwise, it might lead to errors if any of these elements are `null`.